companion-snap-compose/README.md

```diff
@@ -0,0 +1,208 @@
+# LoRa Basic Station Snap
+
+This is a [companion snap](https://ubuntu.com/core/docs/docker-companion-snap) for [LoRa Basics™ Station for Docker](https://github.com/xoseperez/basicstation-docker).
+This companion snap allows running and configuring a LoRa gateway on Ubuntu Core devices.
+
+## Install the snap
+
+> The snap is not yet available from the Snap Store.
+> It is currently required to build it yourself.
+> Refer to [Building the snap](#building-the-snap).
+
+To install the snap from the Snap Store, run:
+
+```
+sudo snap install lora-basicstation
+```
+
+This snap needs to interact with the Docker Engine on the host.
+To allow that, we connect the required docker interfaces:
+
+```
+sudo snap connect lora-basicstation:docker docker:docker-daemon
+sudo snap connect lora-basicstation:docker-executables docker:docker-executables
+```
+
+## Configuration
+
+Configuration of the gateway is done similar to the Docker container, 
+using the same config options as listed in the [upstream documentation](https://github.com/xoseperez/basicstation-docker?tab=readme-ov-file#configure-the-gateway).
+The only difference is that we use snap config options rather than environment variables.
+It would be possible to create an environment variable file and pass that to Docker, but we prefer to use the standard snap configuration options.
+This allows us to leverage the snap ecosystem to maintain and update our Ubuntu Core devices.
+
+To translate from environment variables to snap options, we only need to change the name or key of the configuration option.
+The values remain the same.
+For the variable name, change it to lower case and replace underscores with hyphens.
+Then prefix it with `env.`.
+
+Example:
+`GATEWAY_EUI=DEADFFFEBEEF` becomes `env.gateway-eui=DEADFFFEBEEF`
+
+After you have the configs in this format, you can set them one by one using the `snap set` command:
+
+```
+sudo snap set lora-basicstation env.tc-uri="wss://eu1.cloud.thethings.network:8887"
+sudo snap set lora-basicstation env.tc-key=NNSXS...
+```
+
+Or with a single command:
+
+```
+sudo snap set lora-basicstation \
+    env.gateway-eui=DEADFFFEBEEF \
+    env.model=SX1301 \
+    env.tc-key=NNSXS... \
+    env.tc-uri=wss://eu1.cloud.thethings.network:8887 \
+    env.device=/dev/spidev0.0 \
+    env.interface=SPI \
+    env.spi-speed=200000 \
+    env.use-libgpiod=1 \
+    env.gpio-chip=gpiochip4 \
+    env.reset-gpio=22
+```
+
+An option can be cleared with the `snap unset` command, ex:
+
+```
+sudo snap unset lora-basicstation env.gateway-eui
+```
+
+You can also verify the options with `snap get`.
+For example directly after installation you will see a number of defaults that are already set:
+
+```
+$ sudo snap get lora-basicstation env
+Key            Value
+env.device     /dev/spidev0.0
+env.gpio-chip  gpiochip0
+```
```

Comment:
Where are these defaults set? Not in the install hook.

Reply:
It got dropped in the compose snap. Adding it back in to the install hook.